debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PGTableSchemaBuilder.java

```diff
@@ -0,0 +1,491 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.postgresql;
+
+import java.sql.Types;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+import io.debezium.connector.postgresql.connection.ReplicaIdentityInfo;
+import io.debezium.connector.postgresql.connection.YBReplicaIdentity;
+import io.debezium.relational.*;
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.errors.SchemaBuilderException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.DebeziumException;
+import io.debezium.annotation.Immutable;
+import io.debezium.annotation.ThreadSafe;
+import io.debezium.data.Envelope;
+import io.debezium.data.SchemaUtil;
+import io.debezium.relational.Key.KeyMapper;
+import io.debezium.relational.Tables.ColumnNameFilter;
+import io.debezium.relational.mapping.ColumnMapper;
+import io.debezium.relational.mapping.ColumnMappers;
+import io.debezium.schema.FieldNameSelector.FieldNamer;
+import io.debezium.schema.SchemaNameAdjuster;
+import io.debezium.spi.topic.TopicNamingStrategy;
+import io.debezium.util.Loggings;
+
+/**
+ * Builder that constructs {@link TableSchema} instances for {@link Table} definitions.
+ * <p>
+ * This builder is responsible for mapping {@link Column table columns} to {@link Field fields} in Kafka Connect {@link Schema}s,
+ * and this is necessarily dependent upon the database's supported types. Although mappings are defined for standard types,
+ * this class may need to be subclassed for each DBMS to add support for DBMS-specific types by overriding any of the
+ * "{@code add*Field}" methods.
+ * <p>
+ * See the <a href="http://docs.oracle.com/javase/6/docs/technotes/guides/jdbc/getstart/mapping.html#table1">Java SE Mapping SQL
+ * and Java Types</a> for details about how JDBC {@link Types types} map to Java value types.
+ *
+ * @author Randall Hauch
+ */
+@ThreadSafe
+@Immutable
+public class PGTableSchemaBuilder extends TableSchemaBuilder {
```

Comment:
I don't quite get this. Have we imported this class from upstream or have you written it?

Reply:
This class is imported from upstream, with a few YB specific changes. Have added a note wherever I have made changes.